src/dotnet-uninstall/Shared/Configs/CommandLineConfigs.cs

```diff
@@ -0,0 +1,174 @@
+            "--major-minor",
+            Messages.UninstallMajorMinorOptionDescription,
+            new Argument<string>
+            {
+                Name = Messages.UninstallMajorMinorOptionArgumentName,
+                Description = Messages.UninstallMajorMinorOptionArgumentDescription
+            });
+
+        public static readonly Option UninstallVerbosityOption = new Option(
+            new[] { "--verbosity", "-v" },
+            Messages.UninstallVerbosityOptionDescription,
+            new Argument<string>
+            {
+                Name = Messages.UninstallVerbosityOptionArgumentName,
+                Description = Messages.UninstallVerbosityOptionArgumentDescription
+            });
+
+        public static readonly Option SdkOption = new Option(
+            "--sdk",
+            Messages.SdkOptionDescription);
+
+        public static readonly Option RuntimeOption = new Option(
+            "--runtime",
+            Messages.RuntimeOptionDescription);
+
+        public static readonly IEnumerable<Option> UninstallMainOptions = new Option[]
+        {
+            UninstallAllOption,
+            UninstallAllLowerPatchesOption,
+            UninstallAllButLatestOption,
+            UninstallAllButOption,
+            UninstallAllBelowOption,
+            UninstallAllPreviewsOption,
+            UninstallAllPreviewsButLatestOption,
+            UninstallMajorMinorOption,
+        };
+
+        public static readonly IEnumerable<Option> AuxOptions = new Option[]
+        {
+            UninstallVerbosityOption,
+            SdkOption,
+            RuntimeOption
+        };
+
+        public static readonly RootCommand UninstallRootCommand = new RootCommand(
+            Messages.UninstallNoOptionDescription,
+            argument: new Argument<IEnumerable<string>>
+            {
+                Name = Messages.UninstallNoOptionArgumentName,
+                Description = Messages.UninstallNoOptionArgumentDescription
+            });
+
+        public static readonly Command ListCommand = new Command(
+            "list",
+            Messages.ListCommandDescription);
+
+        static CommandLineConfigs()
+        {
+            UninstallRootCommand.Add(ListCommand);
+
+            foreach (var option in UninstallMainOptions)
+            {
+                UninstallRootCommand.AddOption(option);
+            }
+
+            foreach (var option in AuxOptions)
+            {
+                UninstallRootCommand.AddOption(option);
+                ListCommand.Add(option);
+            }
+
+            ListCommand.Handler = CommandHandler.Create(ExceptionHandler.HandleException(() => ListCommandExec.Execute()));
+            UninstallRootCommand.Handler = CommandHandler.Create(ExceptionHandler.HandleException(() => UninstallCommandExec.Execute()));
+        }
+
+        public static Option GetUninstallMainOptions(this CommandResult commandResult)
```

Comment:
NIT: Should this say GetUninstallMainOption, singular, given that having more than one specified will throw an exception?

Reply:
Thanks for the catch!! It is indeed a typo and I intended to call it `GetUninstallMainOption`

Reply:
Done.